state.go

```diff
@@ -226,6 +227,51 @@ func (p *State) AccountsForNetwork(network config.Network) *accounts.Accounts {
 	return &accs
 }
 
+// AccountByContractName returns the account for a contract by contract name.
+func (p *State) AccountByContractName(contractName string, network config.Network) (*accounts.Account, error) {
+	deployments := p.conf.Deployments.ByNetwork(network.Name)
+	var accountName string
+	for _, d := range deployments {
+		for _, c := range d.Contracts {
+			if c.Name == contractName {
+				accountName = d.Account
+				break
+			}
+		}
+	}
+	if accountName == "" {
+		return nil, fmt.Errorf("contract not found in state")
```

Comment:
maybe add more info to this error

```suggestion
		return nil, fmt.Errorf("deployment of %s not found for network %s", contractName, network.Name)
```

Reply:
done, thanks!